accepted/PSR-6-cache-meta.md

```diff
@@ -303,3 +303,43 @@ _**Note:** Order descending chronologically._
 [1]: https://docs.google.com/spreadsheet/ccc?key=0Ak2JdGialLildEM2UjlOdnA4ekg3R1Bfeng5eGlZc1E#gid=0
 [2]: https://docs.google.com/spreadsheet/ccc?key=0AsMrMKNHL1uGdDdVd2llN1kxczZQejZaa3JHcXA3b0E#gid=0
 [3]: https://docs.google.com/spreadsheet/ccc?key=0AsMrMKNHL1uGdEE3SU8zclNtdTNobWxpZnFyR0llSXc#gid=1
+
+8. Errata
+----------
+
+### 8.1 Handling of incorrect DateTime values in expiresAt()
+
+The `CacheItemInterface::expiresAt()` method's `$expiration` parameter is untyped
+in the interface, but in the docblock is specified as `\DateTimeInterface`.  The
+intent is that either a `\DateTime` or `\DateTimeImmutable` object is allowed.
+However, `\DateTimeInterface` and `\DateTimeImmutable` were added in PHP 5.5, and
+the authors chose not to impose a hard syntactic requirement for PHP 5.5 on the
+specification.
+
+Despite that, implementers MUST accept only `\DateTimeInterface` or compatible types
```

Comment:
I still think that this should also be a `SHOULD`.

Reply:
The problem with accepting other things is that it encourages non-standard behavior.  If a consumer uses a library that also accepts strings, let's say, even though it's type hinting against PSR-6 interfaces, then it's now unwittingly coupled itself to that implementation and created a hidden dependency.  That's what specs like this are intended to not do.

The goal here is to approximate as well as possible if we had targeted PHP 5.5 and just included the type hint in the first place, which is what we should have done in the first place. 🤷‍♂️

Reply:
Hence it's a should. A should is still a strong recommendation. And as said, even with a type declaration, PHP 7.2 allows dropping that.

Reply:
@kelunik that would be implicit due to the language behavior, the next paragraph says

> [...] as if the method was explicitly typed.
> (Note that the variance rules for a typed parameter may vary between language versions.)

... to cover exactly that edge case.